protostar/commands/test/test_command.py

```diff
@@ -38,6 +40,58 @@
 from .test_command_cache import TestCommandCache
 
 
+@dataclass
+class SuccessfulTestMessage(StructuredMessage):
+    summary: TestingSummary
+    test_collector_result: TestCollector.Result
+
+    def format_human(self, fmt: LogColorProvider) -> str:
+        assert False, "Tests should use live logging for the human-readable output"
+
+    def format_dict(self) -> dict:
+        test_cases = 0
+        for test_suite in self.test_collector_result.test_suites:
+            test_cases += len(test_suite.test_cases)
```

Comment:
I wanted to use `sum` + comprehensive list but the linter complained. I guess this is because the comprehensive list allocates an extra list in the memory.